BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/IndoorPositioning.java

```diff
@@ -97,7 +99,20 @@ private void updateLocation() {
         return usableBeacons;
     }
 
+    public static Location walkingSpeedFilter(Location oldLocation, Location newLocation) {
+        double distance = oldLocation.getDistanceTo(newLocation);
+        double angle = oldLocation.getAngleTo(newLocation);
+        if (distance > HUMAN_WALKING_SPEED) {
+            return oldLocation.getShiftedLocation(HUMAN_WALKING_SPEED, angle);
+        } else {
+            return newLocation;
+        }
+    }
+
     private void onLocationUpdated(Location location) {
+        if (lastKnownLocation != null) {
+            location = walkingSpeedFilter(lastKnownLocation, location);
```

Comment:
This would break the library for anyone that uses it for something else than walking...